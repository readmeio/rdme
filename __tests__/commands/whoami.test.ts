import { describe, afterEach, it, expect, beforeAll } from 'vitest';

import pkg from '../../package.json' with { type: 'json' };
import Command from '../../src/commands/whoami.js';
import configStore from '../../src/lib/configstore.js';
import { runCommandAndReturnResult } from '../helpers/oclif.js';

describe('rdme whoami', () => {
  let run: (args?: string[]) => Promise<string>;

  beforeAll(() => {
    run = runCommandAndReturnResult(Command);
  });

  afterEach(() => {
    configStore.clear();
  });

  it('should error if user is not authenticated', () => {
    configStore.delete('email');
    configStore.delete('project');

    return expect(run()).rejects.toStrictEqual(new Error(`Please login using \`${pkg.name} login\`.`));
  });

  it('should return the authenticated user', () => {
    configStore.set('email', 'email@example.com');
    configStore.set('project', 'subdomain');

    return expect(run()).resolves.toBe('You are currently logged in as email@example.com to the subdomain project.');
  });
});
