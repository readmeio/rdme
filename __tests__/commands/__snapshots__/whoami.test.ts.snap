// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`rdme whoami > should error if user is not authenticated 1`] = `
{
  "error": [Error: Please login using \`rdme login\`.],
  "stderr": "",
  "stdout": "",
}
`;

exports[`rdme whoami > should return the authenticated user 1`] = `
{
  "result": "You are currently logged in as email@example.com to the subdomain project.",
  "stderr": "",
  "stdout": "",
}
`;
